### ISSUE TITLE
Scheduled shutdown task fails due to type issue

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2805]
Thanks to user [aymenSo](https://forum.rockstor.com/u/aymenSo), it appears that a scheduled shutdown task fails ultimately due to a type error, which results in a low-level exception during the attempt to execute the `shutdown` program.

this can likely be traced back to this part of the code:

https://github.com/rockstor/rockstor-core/blob/6c86e3858079ce8edc62c62684a7f2cf50fb6d79/src/rockstor/storageadmin/views/command.py#L304-L310

[phillxnet EDIT] ~~As documented in the forum thread, the `delay` variable should be populated with `+3` instead of just `3` following the `shutdown` man page:~~

https://man7.org/linux/man-pages/man8/shutdown.8.html

Should be an easy fix, however, not sure how one would design a test to this (or whether that is necessary) without shutting down the system every time :) (maybe trigger the shutdown and then cancel it within the delay time?).

---

Forum reference: https://forum.rockstor.com/t/error-on-shutdown-task/9313

**Comment by Hooverdan96** [comment]
Based on the referenced issue #2807 this use case could be addressed possibly with changing the `delay` assignment to

```
    if request.auth is not None:
        delay = str(3)

...
```

**Comment by phillxnet** [comment]
I'm working on the linked issue, so will add this change according there. And if all is well we can close this issue as a duplicate.

**Comment by phillxnet** [comment]
I've now split this out into it's own pull request: there were a few comment updates etc and shutdown/reboot specific changes that make it cleaner that way. PR currently under test.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2812
